ext/php8/php8/engine_hooks.c

```diff
@@ -119,7 +119,21 @@ static bool dd_should_trace_helper(zend_execute_data *call, zend_function *fbc,
     HashTable *function_table = NULL;
     bool found = ddtrace_try_find_dispatch(scope, &fname, &dispatch, &function_table);
     if (found && dispatch->options & DDTRACE_DISPATCH_DEFERRED_LOADER) {
-        if (Z_TYPE(dispatch->deferred_load_integration_name) != IS_NULL) {
+        do {
```

Comment:
Why do we need a `do { ... } while (0);` here?

Reply:
It's there for better readability. It allowed me to flatten the `if` blocks because I needed to add one more `if` branch that would further indent code that was already three blocks deep. The `while` block allowed me to skip over this block using `break` so that I didn't have to indent even further for the extra check. I could have used a `goto` instead but I find those harder to read.

Reply:
smart! 😄